packages/block-editor/src/components/colors-gradients/panel-color-gradient-settings.js

```diff
@@ -43,14 +42,14 @@ export const PanelColorGradientSettingsInner = ( {
 	const panelId = useInstanceId( PanelColorGradientSettingsInner );
 	const { batch } = useRegistry();
 	if (
-		isEmpty( colors ) &&
-		isEmpty( gradients ) &&
+		( ! colors || colors.length === 0 ) &&
+		( ! gradients || gradients.length === 0 ) &&
 		disableCustomColors &&
 		disableCustomGradients &&
 		settings?.every(
 			( setting ) =>
-				isEmpty( setting.colors ) &&
-				isEmpty( setting.gradients ) &&
+				( ! setting.colors || setting.colors.length === 0 ) &&
```

Comment:
Colors are expected to be an array in both tests and `theme.json`:

https://github.com/WordPress/gutenberg/blob/553a2c2fecc42947ec0b9b5a87440fd1017cbbde/packages/block-editor/src/components/colors-gradients/test/control.js#L19

https://developer.wordpress.org/themes/advanced-topics/theme-json/#color-palette